libraries/ESP8266mDNS/ESP8266mDNS.cpp

```diff
@@ -539,15 +539,24 @@ void MDNSResponder::_parsePacket(){
     while (numAnswers--) {
       // Read name
       stringsRead = 0;
+      size_t last_bufferpos = 0;
       do {
         tmp8 = _conn_read8();
-        if (tmp8 & 0xC0) { // Compressed pointer (not supported)
-          tmp8 = _conn_read8();
-          break;
-        }
         if (tmp8 == 0x00) { // End of name
           break;
         }
+        if (tmp8 & 0xC0) { // Compressed pointer
+          uint16_t offset = ((((uint16_t)tmp8) & 0xC0) << 8) | _conn_read8();
```

Comment:
1.
Would the resulting offset's boundaries be checked here ?

2.
I don't understand something. If the first byte is 0xca and the second 0xaa, then the offset would be 0xcaaa not 0xaaa 
 like with (((uint16_t)(tmp8 & ~0xC0)) << 8) | _conn_read8() ?